### PR TITLE
[fixed] hanging leaf node connection when account can't be found

### PR DESCRIPTION
**Pull request by matthiashanel**
as a result of the issue, the leaf node connection never got created,
even after the account can be found.

Also tracing account id and name (when available)

Signed-off-by: Matthias Hanel <mh@synadia.com>

```
[48700] 2021/06/08 15:02:46.327026 [INF] 127.0.0.1:4224 - lid:12 - Leafnode connection created for account: ACIB3OAWRMNK34DVSNYVQOEHH5RK7GNPHI7RPYZKVBX4UPD4262H2M2G/EXCACC

[48699] 2021/06/08 15:03:59.174597 [INF] 127.0.0.1:4224 - lid:10 - Leafnode connection closed: Server Shutdown account: ADXKBBHIU3HAY3VXAEETH3XSKQSFJ3YUH7NOBHGS5BN3K4SKNUNE7TUV/TEST
[48699] 2021/06/08 15:03:59.174824 [INF] 127.0.0.1:4224 - lid:11 - Leafnode connection closed: Server Shutdown account: ACIB3OAWRMNK34DVSNYVQOEHH5RK7GNPHI7RPYZKVBX4UPD4262H2M2G/EXCACC
[48699] 2021/06/08 15:03:59.175225 [INF] 127.0.0.1:4224 - lid:12 - Leafnode connection closed: Server Shutdown account: ABDGQHM46B4NBP4YIGKBDASU53TZHWPUVI33H5O3RQUPIGZ2QAFGJQKO/SYS
```

### CODE REVIEW

File: server/accounts.go [hunk 1]
```diff
@@ -2822,6 +2822,16 @@ func (s *Server) UpdateAccountClaims(a *Account, ac *jwt.AccountClaims) {
 	s.updateAccountClaimsWithRefresh(a, ac, true)
 }
 
+func (a *Account) traceLabel() string {
+	if a == nil {
+		return ""
```
Comment (kozlovic) [inline server/accounts.go:2827]: `_EMTPY_`

File: server/accounts.go [hunk 2]
```diff
@@ -2938,13 +2948,13 @@ func (s *Server) updateAccountClaimsWithRefresh(a *Account, ac *jwt.AccountClaim
 	for _, e := range ac.Exports {
 		switch e.Type {
 		case jwt.Stream:
-			s.Debugf("Adding stream export %q for %s", e.Subject, a.Name)
+			s.Debugf("Adding stream export %q for %s", e.Subject, a.traceLabel)
```
Comment (kozlovic) [inline server/accounts.go:2951]: Should `a.traceLabel` not be a function call here? If so, there are several occurrences of that down there.

**Review by kozlovic — CHANGES_REQUESTED** [review]

File: server/accounts.go (hunk 1 above)
Comment (matthiashanel) [reply server/accounts.go]: fixed

File: server/accounts.go (hunk 2 above)
Comment (matthiashanel) [reply server/accounts.go]: fixed

File: server/leafnode.go [hunk 3]
```diff
@@ -818,13 +818,15 @@ func (s *Server) createLeafNode(conn net.Conn, rURL *url.URL, remote *leafNodeCf
 		acc, err = s.LookupAccount(lacc)
 		if err != nil {
 			s.Errorf("No local account %q for leafnode: %v", lacc, err)
+			c.closeConnection(ClientClosed)
```
Comment (kozlovic) [inline server/leafnode.go:821]: Ok, I have found that fits better: `MissingAccount`.

**Review by kozlovic — CHANGES_REQUESTED** [review]

**Pull request merged**